scalyr_agent/configuration.py

```diff
@@ -50,7 +52,75 @@ class Configuration(object):
     This also handles reporting status information about the configuration state, including what time it was
     read and what error (if any) was raised.
     """
-    def __init__(self, file_path, default_paths):
+
+    # Authoritative list of params that can be overridden by environment variables.
+    ENVIRONMENT_AWARE_CONFIG_PARAMS = {
```

Comment:
Do we need this dict?  I think we are going to allow all global parameters to be configured via an environment variable.. so the only useful information here is None vs non-None.  And only one param has a non-None value.   We could modify the `_verify_or_set....` config methods to allow you to override the env var name used for the variable.

Reply:
(related to the above)